src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cameltooling.lsp.internal.parser;
+
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import com.github.cameltooling.lsp.internal.instancemodel.CamelURIInstance;
+import com.github.cameltooling.lsp.internal.instancemodel.YamlDSLModelHelper;
+
+/**
+ * @author Lars Heinemann
+ */
+public class CamelKYamlDSLParser extends ParserFileHelper {
+	
+	public static final String STEP_KEY = "steps:";
+	public static final String URI_KEY = "uri:";
+	public static final String REST_KEY = "rest:";
+	public static final String FROM_KEY = "- from:";
+	public static final String TO_KEY = "- to:";
+
+	@Override
+	public String getCamelComponentUri(String line, int characterPosition) {
+		String camelComponentURI = null;
+		String trLine = line.trim();
+		if (trLine.startsWith(URI_KEY) && URI_KEY.length() < characterPosition) {
+			camelComponentURI = trLine.substring(trLine.indexOf(URI_KEY)+ URI_KEY.length()).trim().replaceAll("\"", "");
```

Comment:
Won't the replaceAll will also replace double-quotes inside camel URI? (need to check tomorrow)
isn't only the first and last that we want to get rid of?

Reply:
I'd say it pretty much depends if there is the possibility of " inside the camel uri. I will check that.

Reply:
the usage of the parser is not fixing the problem of double-quotes inside it because the start character is not found correctly

i tried to call the completion at the end of the to uri
```
# camel-k: language=yaml
- id: "rest"
  group: "routes"
  rest:
    verb: "post"
    uri: "/api/route"
    accepts: "text/plain"
    binding-mode: "off"
    steps:
      - convert-body:
          type: "java.lang.String"
      - to:
          uri: "test:name?delimiter=\"&"
```

Reply:
proposed a fix in this PR https://github.com/lhein/camel-language-server/pull/7

Reply:
closing this conversation because I provided a more generic solution